spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -1019,21 +1015,21 @@ public void testMergeWithMultipleUpdatesForTargetRowSmallTargetLargeSourceNoNotM
     Dataset<Integer> ds = spark.createDataset(sourceIds, Encoders.INT());
     ds.union(ds).createOrReplaceTempView("source");
 
-    String errorMsg = "a single row from the target table with multiple rows of the source table";
-    AssertHelpers.assertThrowsCause(
-        "Should complain about multiple matches",
-        SparkRuntimeException.class,
-        errorMsg,
-        () -> {
-          sql(
-              "MERGE INTO %s AS t USING source AS s "
-                  + "ON t.id > s.value "
-                  + "WHEN MATCHED AND t.id = 1 THEN "
-                  + "  UPDATE SET id = 10 "
-                  + "WHEN MATCHED AND t.id = 6 THEN "
-                  + "  DELETE",
-              commitTarget());
-        });
+    String errorMsg = "MERGE statement matched a single row from the target table with multiple rows of the source table.";
+    Assertions.assertThatThrownBy(
+            () -> {
+              sql(
+                  "MERGE INTO %s AS t USING source AS s "
+                      + "ON t.id > s.value "
+                      + "WHEN MATCHED AND t.id = 1 THEN "
+                      + "  UPDATE SET id = 10 "
+                      + "WHEN MATCHED AND t.id = 6 THEN "
+                      + "  DELETE",
+                  commitTarget());
+            })
+        .cause()
+        .isInstanceOf(SparkRuntimeException.class)
+        .hasMessageContaining(errorMsg);
```

Comment:
should this be `.hasMessage()` now that we're using a more specific error msg?

Reply:
```
[MERGE_CARDINALITY_VIOLATION] The ON search condition of the MERGE statement matched a single row from the target table with multiple rows of the source table.
This could result in the target row being operated on more than once with an update or delete operation and is not allowed.
```
This is the complete error message